test/e2e/storage/testsuites/testdriver.go

```diff
@@ -99,9 +99,42 @@ type DriverInfo struct {
 	RequiredMountOption  sets.String         // Map of string for required mount option (Optional)
 	Capabilities         map[Capability]bool // Map that represents plugin capabilities
 
-	// Parameters below will be set inside test loop by using SetCommonDriverParameters.
-	// Drivers that implement TestDriver is required to set all the above parameters
-	// and return DriverInfo on GetDriverInfo() call.
-	Framework *framework.Framework       // Framework for the test
-	Config    framework.VolumeTestConfig // VolumeTestConfig for thet test
+	Config TestConfig // Test configuration for the current test.
+}
+
+// TestConfig represents parameters that control test execution.
+// They can still be modified after defining tests, for example
+// in a BeforeEach or when creating the driver.
```

Comment:
My understanding is that `TestConfig` is ultimately used to configure each test, but it's passed to the driver during initialization so the driver has a chance to add more configs, is this correct?

It'd be helpful to define even more precisely how `TestConfig` is used here, to make its misuse by future devs less likely.

Reply:
Correct. `DriverInfo` is the static part that gets defined by the test driver author once. `TestConfig` is the part which is more dynamic and contains fields that can be different for different tests.

Reply:
@verult: Does 0fa0ac7ff75987c17a96c62bb4e00c6020cf8cd2 sufficiently explain the relationship between test definition, test execution, DriverInfo and TestConfig?

I am aware that `testsuites.RunTestSuites` is named a bit inappropriately, because it doesn't actually run any tests, it merely defines them. But I didn't want to make more changes than strictly necessary, therefore I have kept the name.

Reply:
Thanks for the doc! Looks great to me. My one nit would be I don't think think it's necessary to describe the ginkgo interactions common across all e2es, but I don't mind either way.

Reply:
Let's keep it in. I'm going to come back to that in the next PR where I introduce delayed test registration, because then it won't be just the usual "register at init".